packages/popover2/src/popover2SharedProps.ts

```diff
@@ -223,7 +240,9 @@ export interface IPopover2SharedProps<TProps> extends OverlayableProps, Props {
      *
      * Mutually exclusive with `children` and `targetTagName` props.
      */
-    renderTarget?: (props: Popover2TargetProps & TProps) => JSX.Element;
```

Comment:
`TProps` was not getting inferred correctly before this change. You can see an example of this in `popover2MinimalExample.tsx`:

```
                <Popover2
                    {...baseProps}
                    minimal={true}
                    renderTarget={({ isOpen, ref, ...p }) => (
                        <Button {...p} active={isOpen} elementRef={ref} intent={Intent.PRIMARY} text="Minimal" />
                    )}
                />
```

here, `p` was getting inferred as an empty object `{}`, which is not what we want. that basically suppresses type checks on that object and allows anything through.

after this change, `p` correctly gets the types of the attributes injected by Popover2 here:

<img width="941" alt="image" src="https://user-images.githubusercontent.com/723999/214108762-42291c90-9ab0-45af-a8ad-314c90ec0b4a.png">